tiledb/sm/buffer/buffer_list.h

```diff
@@ -110,13 +110,13 @@ class BufferList {
   std::vector<Buffer> buffers_;
 
   /** The index of the buffer containing the current global offset. */
-  size_t current_buffer_index_;
+  size_t current_buffer_index_ = 0;
```

Comment:
Let's leave these unassigned in the header. The motivation is 1) to isolate all value assignment to the implementation file and 2) keep it consistent with the rest of the code base.

While we're here, we could make an improvement by list-initializing these values in the constructor rather than assigning them in the body of the constructor.

Reply:
In my idea was here like, if there is one constructor at all, why do we need to define it explicitly? Let compiler do work for you instead of writing it.

However, your concerns are OK for me and I can modify this PR in that way you describe it.

Reply:
That's totally valid for this class in the current state, but that may not always be true if we add more constructors that wish to initialize these variables in a different state. It's a really minor issue, so let's just keep it consistent with the rest of the code base by not value-initializing member variables with their definitions.

Reply:
Now, replaced with list-initialized members in implementation file. Could you please review it again?